src/sidebar/index.js

```diff
@@ -347,13 +355,54 @@ function startAngularApp(config) {
   angular.bootstrap(appEl, ['h'], { strictDi: true });
 }
 
-fetchConfig(appConfig)
-  .then(config => {
-    startAngularApp(config);
-  })
-  .catch(err => {
-    // Report error. This will be the only notice that the user gets because the
-    // sidebar does not currently appear at all if the Angular app fails to
-    // start.
-    console.error('Failed to start Hypothesis client: ', err);
-  });
+// ** Launch the angular app **
+//
+// There are 3 ways to get the config to launch the app.
+//  Case 1: [Classic embed] - From the query string of the embedder frame.
+//  Case 2a: [RPC with unknown parent] - From a ancestor parent frame that passes it down via RPC. (deprecated)
+//  Case 2b: [RPC with known parent] - From a ancestor parent frame that passes it down via RPC.
+
+const hostPageConfig = hostConfig(window);
+const requestConfigFromFrame = hostPageConfig.requestConfigFromFrame;
+const configFrameAncestorLevel = hostPageConfig.configFrameAncestorLevel;
+
```

Comment:
I would prefer that the complete logic for fetching the config could be pushed down into `fetch-config.js` and the logic in `src/sidebar/index.js` is kept as simple as possible. This will make it easier to write tests for this functionality and also help to avoid  `src/sidebar/index.js` growing into a huge module which is difficult to follow.